submissions/salsita/src/reducers/masterReducer.js

```diff
@@ -0,0 +1,50 @@
+import * as Actions from '../constants/actions';
+import * as ScrollingReducer from './scrollingReducer';
+import * as SithLoadingReducer from './sithLoadingReducer';
+import * as PlanetReducer from './planetReducer';
+
+// Master reducer concept inspired by @Retolements
+// http://www.code-experience.com/problems-with-flux/
+export default (reduction, action) => {
+  const type = action.type;
+  const payload = action.payload;
+
+  switch (type) {
+    case Actions.APPLICATION_MOUNTING:
+      return SithLoadingReducer.applicationMounting(reduction, payload);
+    break;
+    case Actions.SITH_LOADING_STARTED:
+      return SithLoadingReducer.sithLoadingStarted(reduction, payload);
+    break;
+    case Actions.SITH_LOADED:
+      return reduction.withMutations(mutableReduction => {
+        mutableReduction.update(r => SithLoadingReducer.sithLoaded(r, payload));
+        mutableReduction.update(r => ScrollingReducer.enableControlButtons(r));
+        mutableReduction.update(r => PlanetReducer.checkObiwanSpottedSith(r));
+      });
+    break;
+    case Actions.SCROLLED_UP:
+      return reduction.withMutations(mutableReduction => {
+        mutableReduction.update(r => ScrollingReducer.scrolledUp(r));
+        mutableReduction.update(r => ScrollingReducer.enableControlButtons(r));
+        mutableReduction.update(r => PlanetReducer.checkObiwanSpottedSith(r));
+      });
+    break;
+    case Actions.SCROLLED_DOWN:
+      return reduction.withMutations(mutableReduction => {
+        mutableReduction.update(r => ScrollingReducer.scrolledDown(r));
```

Comment:
Maybe I'm just a DRY fanatic (aka lazy) but I would probably have said:

```
case Actions.SCROLLED_UP:
case Actions.SCROLLED_DOWN:
   ...
   mutableReduction.update(r => ScrollingReducer[type === Actions.SCROLLED_UP ? 'scrolledUp' : 'scrolledDown'](r));
  ...
```

This would make maintenance easier if you change the way scrolling is handled since you won't have to make the changes in two places.

BTW: in case you think this is too much, I almost considered ```
ScrollingReducer[`scrolled${type === Actions.SCROLLED_UP ? 'Up' : 'Down'}`]
```
:wink: